### PR TITLE
Initial refactor of CI for LTS release

### PR DESCRIPTION
**Pull request by TravisEz13**
## PR Summary

Initial refactor of CI for LTS release.

TODO:
 - [ ] Release build will have to be refactored after this.
 - [ ] Add back community stable.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
  - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `Dockerfile`, `.sh`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
  - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **Adds a new image or tag**
  - [x] Not Applicable
- **OR**
  - [ ] Update [README.powershellcommunity.md](https://github.com/PowerShell/PowerShell-Docker/blob/master/assets/README.powershellcommunity.md)
  - [ ] Update [vsts-ci.yml](https://github.com/PowerShell/PowerShell-Docker/blob/master/vsts-ci.yml)

### CODE REVIEW

**Comment by TravisEz13** [comment]
ACR CI build is here:  https://dev.azure.com/powershell/docker/_build/results?buildId=43082&view=logs&j=764f8789-4dd5-52ca-5596-2584750ea9ac&t=764f8789-4dd5-52ca-5596-2584750ea9ac

File: build.ps1 [hunk 1]
```diff
@@ -448,6 +493,50 @@ End {
...
+            $channelGroups = $tagGroups.$repo | Group-Object -Property Channel
+            foreach($channelGroup in $channelGroups)
+            {
+                $channelName = $channelGroup.Name
+                Write-Verbose "generating $channelName json"
+                $osGroups = $channelGroup.Group | Group-Object -Property os
+                foreach ($osGroup in $osGroups) {
+                    $osName = $osGroup.Name
+
+                    foreach ($tag in $osGroup.Group | Where-Object { $_.Name -notlike '*/*' } | Sort-Object -Property dockerfile) {
```
Comment (adityapatwardhan) [inline build.ps1]: Add a comment for filtering sub images

File: build.ps1 [hunk 2]
```diff
@@ -448,6 +493,50 @@ End {
...
+                        }
+                    }
+                }
+            }
+        }
+
+        foreach ($channelName in $matrix.Keys) {
+            foreach ($osName in $matrix.$channelName.Keys) {
+                $osMatrix = $matrix.$channelName.$osName
+                $yaml = $osMatrix | ConvertTo-Json -Compress
```
Comment (adityapatwardhan) [inline build.ps1]: rename yaml to json

**Comment by TravisEz13** [comment]
I verified this did not affect the release build.  I'm going to merge LTS into master.

**Pull request merged**